dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/UnifiedClassLoader.java

```diff
@@ -112,29 +112,49 @@ public UnifiedClassLoader(ClassLoader parent, ClassLoader... followOns) {
     @Trivial
     @FFDCIgnore(ClassNotFoundException.class)
     Class<?> loadClass0(String name, boolean resolve) throws ClassNotFoundException {
-        try {
-            return super.loadClass(name, resolve);
-        } catch (ClassNotFoundException ex) {
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "CNFE from super classloader " + super.toString(), ex);
+        Object lock = getClassLoadingLock(name);
+        synchronized (lock) {
+            Class<?> result = findLoadedClass(name);
+            if (result != null) {
+                return result;
+            }
+            ClassLoader parent = getParent();
```

Comment:
Later in the method they do a doPriv for getParent.  Do we need to do the same here?  Of course this can be avoided by storing the parent in the class like what is done with the other implementations.

Also, there is no handling of the parent being null.

Reply:
I agree:

- need to handle null parent (although I don't think this loader will ever have a null parent.
- should store parent in an accessible field to avoid security check locally.